serenity-jira-plugin/src/main/java/net/serenitybdd/plugins/jira/service/JiraIssueTracker.java

```diff
@@ -122,58 +115,40 @@ public void updateComment(IssueComment issueComment) {
      * @throws IssueTrackerUpdateException
      */
     public String getStatusFor(final String issueKey) throws IssueTrackerUpdateException {
-        throw new NotImplementedException("TODO");
-//        String status = null;
-//        try {
-//            String token = jiraConnection.getAuthenticationToken();
-//
-//            RemoteIssue issue = jiraConnection.getJiraSoapService().getIssue(token, issueKey);
-//            checkThatIssueExists(issue, issueKey);
-//            status = getStatusLabel(issue);
-//
-//        } catch (IOException e) {
-//            processJiraException(issueKey, e);
-//        }
-//        return status;
+        try {
+            Optional<IssueSummary> issue = jiraConnection.getRestJiraClient().loadByKey(issueKey);
+            return issue.get().getStatus();
```

Comment:
getRestJiraClient() returns an Optional<>, but then you just use the get() assuming that the issue has been found. So the exception will be odd. It would probably make more sense to throw some special exception (e.g. IssueNotFoundException) in that case to make it easier to troubleshoot.